pkg/controller/account/account_controller.go

```diff
@@ -646,10 +638,24 @@ func (r *ReconcileAccount) initializeRegions(reqLogger logr.Logger, currentAcctI
 		AllRegions: aws.Bool(false),
 	})
 	if err != nil {
+		// Retry on failures related to the slow AWS API
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == "OptInRequired" {
+				return nil
+			}
+		}
 		reqLogger.Error(err, "Failed to retrieve list of regions enabled in this account.")
 		return err
 	}
 
+	// We're about to kick off region init in a goroutine. This status makes subsequent
+	// Reconciles ignore the Account (unless it stays in this state for too long).
+	utils.SetAccountStatus(currentAcctInstance, "Initializing Regions", awsv1alpha1.AccountInitializingRegions, AccountInitializingRegions)
```

Comment:
What is the reason to move the status update to after the DescribeRegions call? If the DescibeRegions call fails what is the status of the account? Will the operator retry initializing this account?

Reply:
If the describe regions call fails we want to reconcile and that is why I moved the part where we set the account status to after the describeRegions call. That way if the describeRegions call fails with an "OptInRequired" error we just reconcile and try again. I was considering failing the account if the call failed for any other reasons but decided against it but looking at it again maybe we should?

Reply:
I didn't want to fail an account that could potentially self resolve I guess? But to answer your question better, yes it just tries to initialize the region again which it wouldn't if we updated the status sooner.

Reply:
IMO moving the status update to after the describe-regions makes sense, the logic should be able to catch up and try again calling initializeRegions.

The potential issue I see is if AWS never recovers from the OptInRequired state for the describe-regions call. It will loop indefinitely and would never mark the account as Failed. But I guess short term we don't have a better option.

Reply:
Makes sense for it to try again thanks.